components/raftstore/src/store/peer.rs

```diff
@@ -4443,7 +4454,10 @@ pub trait RequestInspector {
             if apply::is_conf_change_cmd(req) {
                 return Ok(RequestPolicy::ProposeConfChange);
             }
-            if get_transfer_leader_cmd(req).is_some() {
+            if get_transfer_leader_cmd(req).is_some()
+                && !WriteBatchFlags::from_bits_truncate(req.get_header().get_flags())
```

Comment:
I don't get it. Isn't propose transfer leader after propose locks removal enough? Why need to persist the transfer leader command and apply?

Reply:
Sorry, perhaps the explanation in the PR description is not clear enough.

I was afraid that the proposal containing all pessimistic locks is too large. That proposal may take a bit long time to apply. So, if the real transfer leader happens after the transferree applies the pessimistic locks, the gap will be much smaller. Then, the unavailable duration won't increase compared to when there're no in-memory pessimistic locks.

Reply:
Perhaps you can use a callback instead of issuing an actual write.